integration-tests/tests/src/node/clean-exit.ts

```diff
@@ -0,0 +1,31 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2022 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import { execSync } from "child_process";
+
+describe("Clean exit for Node.js scripts", () => {
+  // Repro for https://github.com/realm/realm-js/issues/4535 - currently still failing
+  it.skip("exits cleanly when creating a new Realm.App", () => {
+    execSync(
+      `node -e 'const Realm = require("realm"); const app = new Realm.App({ id: "myapp-abcde" }); Realm.clearTestState();'`,
+      {
+        timeout: 5000,
```

Comment:
I think this extends the timeout of the test? I.e. if it takes 3 seconds for the process to exit, it will fail although it didn't timeout. Could we read this timeout from the test's context instead?
```suggestion
        timeout: this.timeout,
```
```suggestion
        timeout: 5000,
```

Reply:
Ah yeah, good spot. How about this change: https://github.com/realm/realm-js/pull/4556/commits/924c7dbc6e1039999bde00b1df889ffd10692971 so that it never exceeds either the test timeout or 5000ms? 

I think 5000ms should always be enough for this test in any case, if you were running the suite with say 60 second timeout for some reason, there's no point in this one waiting for 60 seconds also.

Reply:
Good idea 👍 Looks great.